src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/storm/bolt/speaker/SpeakerRouter.java

```diff
@@ -113,10 +126,12 @@ private void speakerMessage(Tuple input, Message message) throws PipelineExcepti
     }
 
     private void proxySpeaker(Tuple input, Message message) throws PipelineException {
-        if (message instanceof InfoMessage) {
-            proxySpeaker(input, ((InfoMessage) message).getData());
-        } else {
-            log.error("Do not proxy speaker message - unexpected message type \"{}\"", message.getClass());
+        if (active) {
```

Comment:
I would like to put this filter "closer" to the `handleInput`... We can wrap the whole `handleInput` in this condition if we move zookeeper stream routing inside the `dispatch` method... or allow AbstractBolt to do it for us.

Reply:
We can not wrap the whole `handleInput` in this condition, because sometimes we wait for responses and have to handle them.